.github/actions/save_dbt_cache/action.yaml

```diff
@@ -0,0 +1,23 @@
+name: Save dbt cache
+description: Updates dbt cache using S3 storage.
+inputs:
+  path:
+    description: The local path to the state dir to upload as the new cache.
+    required: true
+  key:
+    description: The key to use for the cache.
+    required: true
+  bucket:
+    description: The S3 bucket that should store the cache.
+    required: false
+    default: ccao-dbt-cache-us-east-1
+runs:
+  using: composite
+  steps:
+    - name: Save dbt cache
+      run: aws s3 cp "$CACHE_PATH/manifest.json" "s3://$BUCKET/$KEY/manifest.json"
```

Comment:
This remote storage naming mostly works well, but in pull requests where the branch name includes a slash, it results in some potentially confusing [nested folder behavior](https://github.com/ccao-data/data-architecture/actions/runs/5968306499/job/16194026297#step:10:59):

```
upload: dbt/target/manifest.json to s3://ccao-dbt-cache-us-east-1/dbt-cache-jeancochrane/89-fix-broken-cache-restoration/manifest.json
```

My guess is that this is probably fine, since the cache bucket will mostly be used by bots and it might even be nice to have higher-level organization (e.g. anything in the `dbt-cache-jeancochrane` subfolder likely corresponds to a cache for a branch that I authored). 

I also wonder if we want to add a [lifecycle configuration](https://docs.aws.amazon.com/AmazonS3/latest/userguide/how-to-set-lifecycle-configuration-intro.html) to this bucket? It'll need a bit of research, since we'll want the `dbt-cache-master` subfolder to persist indefinitely while everything else can be deleted after a period of inactivity. Curious what you think about all this!

Reply:
IMO this is totally fine, since none of us are likely to use this bucket manually. 

**suggestion (non-blocking):** Let's definitely set some lifecycle rules though just to keep things tidy (likewise on the dbt Athena results buckets if there aren't any already 👀). IMO 60 days to mark all non-master keys for deletion with another 30 days to delete is sufficient. Let me know if you want to set it up or if I should grab it.

Reply:
Cool! I had to adjust the cache keys to create a consistent prefix for PR caches that we could use in the lifecycle policy (`pr-caches/`), but I did that in 74031b6b476244103ff676916b803e603c885e26 and added a policy. Does this look right to you? The difference between expiration and deletion is a bit over my head, so perhaps it needs a little tweaking:

![image](https://github.com/ccao-data/data-architecture/assets/14170650/7b900299-8ea0-4c41-b349-2f3fce18b084)
 
Luckly, dbt-athena creates the query results bucket with an existing lifecycle policy that deletes all query results after 3 days -- nice!

![Screenshot 2023-08-25 135118](https://github.com/ccao-data/data-architecture/assets/14170650/092b2066-05f6-4f64-9114-166b40a4f6d1)

Reply:
@jeancochrane That policy looks good to me! 

`ccao-athena-results-us-east-1` is the bucket I setup for general Athena output. I also setup that lifecycle rule. I was talking about `ccao-dbt-athena-ci-us-east-1` and `ccao-dbt-athena-test-us-east-1`, neither of which have lifecycle rules. IMO we probably should setup lifecycle rules for those as well.

Reply:
Oops, my bad @dfsnow, I totally misread the bucket name 🤦🏻‍♀️ I went ahead and copied that policy to `ccao-dbt-athena-ci-us-east-1` and `ccao-dbt-athena-test-us-east-1`!